crypto/s2n_certificate.c

```diff
@@ -60,9 +59,11 @@ int s2n_create_cert_chain_from_stuffer(struct s2n_cert_chain *cert_chain_out, st
         GUARD(s2n_alloc(&mem, sizeof(struct s2n_cert)));
         new_node = (struct s2n_cert *)(void *)mem.data;
 
-        GUARD(s2n_alloc(&new_node->raw, s2n_stuffer_data_available(&cert_out_stuffer)));
+        if (s2n_alloc(&new_node->raw, s2n_stuffer_data_available(&cert_out_stuffer))) {
```

Comment:
Does it make sense to put a `DEFER_CLEANUP` on the `struct s2n_blob mem`, preventing the multiple `s2n_free` calls below?

Reply:
I'm pretty sure that the ```struct s2n_blob mem``` points to memory that needs to be used after the function exits. So using DEFER_CLEANUP on it would cause the memory to be freed too early. But I also could be reading this function wrong, it's a little difficult to see where mem eventually ends up.